src/cluster/cluster_defs.h

```diff
@@ -33,11 +34,36 @@ enum {
 inline constexpr const char *errInvalidNodeID = "Invalid cluster node id";
 inline constexpr const char *errInvalidSlotID = "Invalid slot id";
 inline constexpr const char *errSlotOutOfRange = "Slot is out of range";
+inline constexpr const char *errSlotRangeInvalid = "Slot range is invalid";
 inline constexpr const char *errInvalidClusterVersion = "Invalid cluster version";
 inline constexpr const char *errSlotOverlapped = "Slot distribution is overlapped";
 inline constexpr const char *errNoMasterNode = "The node isn't a master";
 inline constexpr const char *errClusterNoInitialized = "The cluster is not initialized";
 inline constexpr const char *errInvalidClusterNodeInfo = "Invalid cluster nodes info";
 inline constexpr const char *errInvalidImportState = "Invalid import state";
 
-using SlotRange = std::pair<int, int>;
+struct SlotRange {
+  SlotRange(int start, int end) : start(start), end(end) {}
+  SlotRange() : start(-1), end(-1) {}
+  bool IsValid() const {
+    return start >= 0 && start < kClusterSlots && end >= 0 && end < kClusterSlots && start <= end;
+  }
+
+  bool Contain(int slot) const { return IsValid() && slot >= start && slot <= end; }
```

Comment:
The semantics of range is preferably `[start,end)`.

Reply:
I referenced the `SetSlotRanges` implementation, and it should be following `[start, end]`.
https://github.com/apache/kvrocks/blob/7ed512e835201abe3f68f3c795dabb13acdc6733/src/cluster/cluster.cc#L120
Should they be harmonized, especially since `migrate` and `setslot` tend to appear together when the user is using them.
If you change it to `[start, end)` then `SetSlotRanges` should be changed synchronously too, what do you think to do?

Reply:
OK, keep it that way.